app/service/login_handlers/default.py

```diff
@@ -58,6 +58,8 @@ async def _ldap_login(self, username, password):
         dn = self._ldap_config.get('dn')
         user_attr = self._ldap_config.get('user_attr') or 'uid'
         user_string = '%s=%s,%s' % (user_attr, username, dn)
+        user_format_string = self._ldap_config.get("user_format") or "{user_attr}={user},{dn}"
+        user_string = user_format_string.format(user_attr=user_attr, user=username, dn=dn)
```

Comment:
```suggestion
        user_format_string = self._ldap_config.get("user_format") or "{user_attr}={user},{dn}"
        try:
            user_string = user_format_string.format(user_attr=user_attr, user=username, dn=dn)
        except KeyError:
            user_string = '%s=%s,%s' % (user_attr, username, dn)
```
In case users provide a `user_format` value in their LDAP config that doesn't contain the required parameters, we should default to the previous format.